### DISCUSSION TITLE
Cannot extend sdk 6.0 image with more package installations

### DISCUSSION THREAD

**Discussion by icelava** [dotnet/dotnet-docker#5617] [General]
Am trying to customise the sdk:6.0 image by installing more software

```
FROM mcr.microsoft.com/dotnet/sdk:6.0
RUN apt update && apt install xq
```

But it seems like the particular software xq is not available to apt

>  => ERROR [2/4] RUN apt update && apt install xq                                                                                                                                                                             3.6s
> ------
>  > [2/4] RUN apt update && apt install xq:
> 0.283 
> 0.283 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
> 0.283 
> 0.355 Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
> 0.368 Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
> 0.379 Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
> 0.450 Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8068 kB]
> 0.698 Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [276 kB]
> 0.714 Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [18.8 kB]
> 1.567 Fetched 8570 kB in 1s (6732 kB/s)
> 1.567 Reading package lists...
> 1.970 Building dependency tree...
> 2.088 Reading state information...
> 2.098 All packages are up to date.
> 2.103
> 2.103 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
> 2.103
> 2.107 Reading package lists...
> 2.516 Building dependency tree...
> 2.630 Reading state information...
> 2.751 E: Unable to locate package xq
> ------
> Dockerfile_sdk_6_0_ci:2
> --------------------
>    1 |     FROM mcr.microsoft.com/dotnet/sdk:6.0
>    2 | >>> RUN apt update && apt install xq

Thing is, this works just fine with the _sdk:8.0_ image.

I'm not familiar with the underlying distributions for the dotnet base images. Any idea on why the difference in behaviour?

**Comment by mthalman** [comment]
`6.0` == Debian 11 (Bullseye)
`8.0` == Debian 12 (Bookworm)

It looks like the `xq` package is only available on Bookworm and higher: https://packages.debian.org/bookworm/xq

If you would like to use Bookworm for .NET 6, you can use this tag: `6.0-bookworm-slim`.

You can read more about supported platforms for .NET container images at https://github.com/dotnet/dotnet-docker/blob/main/documentation/supported-platforms.md.

**Reply by icelava** [reply]
Thanks for the info. 6.0-bookworm-slim works.
